src/OpenCV/Features2d.hsc

```diff
@@ -285,6 +300,201 @@ orbDetectAndCompute orb img mbMask = unsafeWrapException $ do
+       --   'minConvexity' (inclusive) and 'maxConvexity' (exclusive).
+     , blob_filterByInertia :: !(Maybe BlobFilterByInertia)
+       -- ^ Extracted blobs have this ratio between 'minInertiaRatio' (inclusive)
+       --   and 'maxInertiaRatio' (exclusive).
+     }
+
+defaultSimpleBlobDetectorParams :: SimpleBlobDetectorParams
+defaultSimpleBlobDetectorParams =
+    SimpleBlobDetectorParams
+    { blob_minThreshold = 50
+    , blob_maxThreshold = 220
+    , blob_thresholdStep = 10
+    , blob_minRepeatability = 2
+    , blob_minDistBetweenBlobs = 10
+    , blob_filterByArea = Just (BlobFilterByArea 25 5000)
+    , blob_filterByCircularity = Nothing
+    , blob_filterByColor = Just (BlobFilterByColor 0)
+    , blob_filterByConvexity = Just (BlobFilterByConvexity 0.95 (1/0))
+    , blob_filterByInertia = Just (BlobFilterByInertia 0.1 (1/0))
+    }
+
+--------------------------------------------------------------------------------
+
+newSimpleBlobDetector :: SimpleBlobDetectorParams -> IO SimpleBlobDetector
+newSimpleBlobDetector SimpleBlobDetectorParams{..} = fromPtr
+    [CU.block|Ptr_SimpleBlobDetector * {
+      cv::SimpleBlobDetector::Params params;
+      params.blobColor           = $(unsigned char c'blobColor);
+      params.filterByArea        = $(bool c'filterByArea);
+      params.filterByCircularity = $(bool c'filterByCircularity);
+      params.filterByColor       = $(bool c'filterByColor);
+      params.filterByConvexity   = $(bool c'filterByConvexity);
+      params.filterByInertia     = $(bool c'filterByInertia);
+      params.maxArea             = $(float c'maxArea);
+      params.maxCircularity      = $(float c'maxCircularity);
+      params.maxConvexity        = $(float c'maxConvexity);
+      params.maxInertiaRatio     = $(float c'maxInertiaRatio);
+      params.maxThreshold        = $(float c'maxThreshold);
+      params.minArea             = $(float c'minArea);
+      params.minCircularity      = $(float c'minCircularity);
+      params.minConvexity        = $(float c'minConvexity);
+      params.minDistBetweenBlobs = $(float c'minDistBetweenBlobs);
+      params.minInertiaRatio     = $(float c'minInertiaRatio);
+      params.minRepeatability    = $(float c'minRepeatability);
+      params.minThreshold        = $(float c'minThreshold);
+      params.thresholdStep       = $(float c'thresholdStep);
+      cv::Ptr<cv::SimpleBlobDetector> detectorPtr =
+        cv::SimpleBlobDetector::create(params);
+      return new cv::Ptr<cv::SimpleBlobDetector>(detectorPtr);
+    }|]
+  where
+    c'minThreshold        = realToFrac blob_minThreshold
+    c'maxThreshold        = realToFrac blob_maxThreshold
+    c'thresholdStep       = realToFrac blob_thresholdStep
+    c'minRepeatability    = realToFrac blob_minRepeatability
+    c'minDistBetweenBlobs = realToFrac blob_minDistBetweenBlobs
+    c'filterByArea        = fromBool (blob_filterByArea /= Nothing)
```

Comment:
How about using `isJust :: Maybe a -> Bool` from `Data.Maybe` instead of `/= Nothing`? Might be a bit more obvious.

`fromBool (isJust blob_filterByArea`)